connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -229,6 +230,9 @@ public void destroyConnector(final @PathParam("connector") String connector,
                     throw new ConnectRestException(Response.Status.CONFLICT.getStatusCode(),
                             "Cannot complete request because of a conflicting operation (e.g. worker rebalance)");
                 }
+            } else if (cause instanceof StaleConfigException) {
+                throw new ConnectRestException(Response.Status.CONFLICT.getStatusCode(),
+                        "Cannot complete request because of a conflicting operation (e.g. worker rebalance)");
```

Comment:
Should we preserve the error message? Otherwise the user can't actually tell why this happened, they'll see the same error message for both stale config and forwarding errors. It's pretty important to expose enough info to tell what's going on because in cluster mode the user making the request may not have easy access to the worker logs.